### PR TITLE
More AI holograms

### PR DESCRIPTION
**Pull request by ghost**
Redone the AI hologram menu, more holograms to choose from as AI (animals/pets). 
Code from Paradise.

:cl:
add: More AI holograms!
/:cl:

### CODE REVIEW

**Comment by jughu** [comment]
Selects I AM THE LAW I AM THE LAW.I AM THE LAW

**Comment by Dorsisdwarf** [comment]
This PR is not atomic. Please seperate the changes to AI holograms and Sechailers.

**Comment by uraniummeltdown** [comment]
Even though it's non-atomic, the changes themselves are very minor QoL and none of them affect balance. Is it absolutely necessary to split this PR?

**Comment by Dorsisdwarf** [comment]
I guess technically not. @Cheridan do you care enough about the ROOLS?

File: code/modules/clothing/masks/hailer.dm [hunk 1]
```diff
@@ -39,28 +41,161 @@
...
 		halt()
-	else
+	else if(actiontype == /datum/action/item_action/adjust)
 		adjustmask(user)
+	else if(actiontype == /datum/action/item_action/selectphrase)
+		switch(aggressiveness)
+			if(1)
+				switch(phrase)
+					if(1)
+						user << "<span class='notice'>You set the restrictor to: Stop in the name of the Law.</span>"
```
Comment (JJRcop) [inline code/modules/clothing/masks/hailer.dm]: Ahhhh make this a list not a long switch seriously what are you doing

**Comment by uraniummeltdown** [comment]
I'll look into trying to make it a list

**Comment by optimumtact** [comment]
@Dorsisdwarf the rule is more to prevent people including unrelated unpopular changes with their pr's.

In cases like this where it's unlikely anyone will mind we usually let it slide.

**Comment by KingPhilipIII** [comment]
I look forward to constantly saying "You have the right to shut the fuck up."

**Comment by uraniummeltdown** [comment]
Does it have to be a list? It's way simpler to leave it as a switch, can't really make it work as a list.

**Comment by A-t48** [comment]
Add cute Korean AI faces too, please. They are in the code already!

**Comment by RemieRichards** [comment]
@A-t48 they're not actually, I could cough up the .dmi but ~reference policy~

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -633,41 +633,95 @@ var/list/ai_list = list()
...
-
-		if(personnel_list.len)
-			input = input("Select a crew member:") as null|anything in personnel_list
-			var/icon/character_icon = personnel_list[input]
-			if(character_icon)
-				qdel(holo_icon)//Clear old icon so we're not storing it in memory.
-				holo_icon = getHologramIcon(icon(character_icon))
+	switch(alert("Would you like to select a hologram based on a crew member, an animal, or switch to a unique avatar?",,"Crew Member","Unique","Animal"))
+		if("Crew Member")
+			var/personnel_list[] = list()
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: don't define byond lists like this.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 3]
```diff
@@ -633,41 +633,95 @@ var/list/ai_list = list()
...
+				input = input("Select a crew member:") as null|anything in personnel_list
+				var/icon/character_icon = personnel_list[input]
+				if(character_icon)
+					qdel(holo_icon)//Clear old icon so we're not storing it in memory.
+					holo_icon = getHologramIcon(icon(character_icon))
+			else
+				alert("No suitable records found. Aborting.")
+
+		if("Animal")
+			var/icon_list[] = list(
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: don't define byond lists like this.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 4]
```diff
@@ -633,41 +633,95 @@ var/list/ai_list = list()
...
+			"eldritch",
+			"ancient machine"
+			)
+
+			input = input("Please select a hologram:") as null|anything in icon_list
+			if(input)
+				qdel(holo_icon)
+				switch(input)
+					if("default")
+						holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo1"))
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: rename their icons to the text used here, and then simply do:
```holo_icon = getHologramIcon(icon('icons/mob/AI.dmi', input))```

Same for the other switches.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 5]
```diff
@@ -633,41 +633,89 @@ var/list/ai_list = list()
...
+			"xeno queen",
+			"horror"
+			)
+
+			input = input("Please select a hologram:") as null|anything in icon_list
+			if(input)
+				qdel(holo_icon)
+				switch(input)
+					if("default")
+						holo_icon = getHologramIcon(icon('icons/mob/AI.dmi', input))
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: as you can see, all 4 cases are now identical, so the switch isn't needed at all (which was my initial point)

**Comment by ExcessiveUseOfCobblestone** [comment]
can you post pics of the new holograms?

**Comment by uraniummeltdown** [comment]
@ExcessiveUseOfCobblestone they're the normal animal sprites run through the hologram proc, there's also a small Nar'Sie hologram

**Comment by coiax** [comment]
Once this is merged, I have an idea for AI holograms.

**Comment by Dorsisdwarf** [comment]
How big is the "small narSie hologram"? IIRC even the small sprite for narsie is like 4x4

**Comment by uraniummeltdown** [comment]
@Dorsisdwarf 
narsieholo

**Comment by optimumtact** [comment]
Is this ready for review?

**Comment by uraniummeltdown** [comment]
@optimumtact yes

**Comment by Shadowlight213** [comment]
>fails travis

**Comment by uraniummeltdown** [comment]
I have no idea why Travis is failing, it compiles and works fine

**Comment by jughu** [comment]
Travis is corrupt and doesn't want this kill travis

**Comment by Cyberboss** [comment]
Read the logs you nerd:

```
code/modules/clothing/masks/hailer.dm:67:error: actiontype: undefined var
code/modules/clothing/masks/hailer.dm:69:error: actiontype: undefined var
```

File: code/modules/clothing/masks/hailer.dm [hunk 6]
```diff
@@ -39,28 +41,161 @@
...
 	else
 		..()
 
 /obj/item/clothing/mask/gas/sechailer/ui_action_click(mob/user, actiontype)
 	if(actiontype == /datum/action/item_action/halt)
 		halt()
-	else
+	else if(actiontype == /datum/action/item_action/adjust)
 		adjustmask(user)
+	else if(actiontype == /datum/action/item_action/selectphrase)
```
Comment (Cyberboss) [inline code/modules/clothing/masks/hailer.dm]: Wait this actually makes no sense, why isn't it errorring on line 65??

**Review by Cyberboss — COMMENTED** [review]
Read the logs you nerd:
```
code/modules/clothing/masks/hailer.dm:67:error: actiontype: undefined var
code/modules/clothing/masks/hailer.dm:69:error: actiontype: undefined var
```

**Comment by uraniummeltdown** [comment]
I'll fix it tomorrow

**Comment by uraniummeltdown** [comment]
Alright it works now

File: code/modules/clothing/masks/hailer.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/clothing/masks/hailer.dm]: This would be much less dense as a list

Comment (optimumtact) [inline code/modules/clothing/masks/hailer.dm]: @uraniummeltdown is there anything you can do to make this code less long and nested

Comment (optimumtact) [inline code/modules/clothing/masks/hailer.dm]: Please I beg of you

File: code/modules/clothing/masks/hailer.dm [hunk 7]
```diff
@@ -39,28 +41,161 @@
 	if(istype(W, /obj/item/weapon/screwdriver))
 		switch(aggressiveness)
 			if(1)
-				user << "<span class='notice'>You set the restrictor to the middle position.</span>"
+				user << "<span class='notice'>You set the aggressiveness restrictor to the second position.</span>"
```
Comment (AnturK) [inline code/modules/clothing/masks/hailer.dm]: Making it just Level X would save the whole switch. Alternatively moving ordinal numbers generation to some helper, could be useful elsewhere.

**Comment by optimumtact** [comment]
Closing until owner shows backup, ping me to reopen.

**Comment by uraniummeltdown** [comment]
@optimumtact Forgot all about this PR, I removed the Sechailer stuff as I prefer random phrases now, it's just the extra AI holos now.

**Comment by AnturK** [comment]
Thanks Jordie

**Pull request merged**